### ISSUE TITLE
need better logging in networking library when black hole clusters are created

### ISSUE THREAD

**Issue by esnible** [istio/istio#5514]
If I create a VirtualService

```
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  name: guestbook
spec:
  gateways:
  - guestbook-gateway
  hosts:
  - '*'
  http:
  - match:
    - uri:
        prefix: /hello
    route:
    - destination:
        host: helloworld-service
```

the Envoy configuration generated by Pilot points to the BlackHoleCluster:

```
                      "name": "guestbook:80",
                      "routes": [
                        {
                          "decorator": {
                            "operation": "guestbook"
                          },
                          "match": {
                            "prefix": "/hello"
                          },
                          "route": {
                            "cluster": "BlackHoleCluster",
                            "timeout": "0.000s",
                            "use_websocket": false
                          }
                        }
                      ]
```

If I add 

```
        port:
          number: 8080
```

to the _destination_ the Envoy config route correctly changes to 

```
                          "route": {
                            "cluster": "outbound|http||helloworld-service.default.svc.cluster.local",
                            "timeout": "0.000s",
                            "use_websocket": false
                          }
```

For this experiment, helloworld-service has a single Port:

```
  ports:
  - name: http
    nodePort: 32262
    port: 8080
    protocol: TCP
    targetPort: 8080
```

According to the documentation https://preliminary.istio.io/docs/reference/config/istio.networking.v1alpha3.html#Destination "Many services only expose a single port or label ports with the protocols they support, in these cases it is not required to explicitly select the port."

**Comment by frankbu** [comment]
The only thing that looks different from the bookinfo services (which work without a port selection) is the protocol and targetPort fields. Can you try to remove them from the service spec and see if that makes a difference?
```
  ports:
  - name: http
    port: 8080
```

**Comment by vadimeisenbergibm** [comment]
I think this is by design. A service can be provided on multiple ports, e.g. 80 and 443. The virtual service, can, for example, rewrite the destination port. Consider a scenario where the user sends a request to `helloworld-service:80` and the virtual service will forward the request to `helloworld-service:443`. So the destination ports are meaningful - a virtual service should know which port to forward to.

**Comment by esnible** [comment]
@vadimeisenbergibm I have three concerns.  First, the documentation https://preliminary.istio.io/docs/reference/config/istio.networking.v1alpha3.html#Destination says "Many services only expose a single port ... in these cases it is *not required* to explicitly select the port."  That documentation is not always correct.

@frankbu and I discovered that if the Gateway port has the same number as the Service port it works as expected without an explicit port number on the Virtual Service.  That isn't obvious.

Finally, Pilot's logging is poor.  I would like at least an INFO-level log that is very explicit: "Creating BlackHoleCluster because of VirtualService default/xyz and no matching <whatever>"  Currently there are several places where the BlackHoleCluster is generated and there is nothing obvious in the logs.

**Comment by rshriram** [comment]
Possible Duplicate of #5016

**Comment by rshriram** [comment]
Addressed by #5543
